src/main/java/com/conveyal/datatools/manager/models/Deployment.java

```diff
@@ -436,19 +438,43 @@ private <O extends Serializable> String writeToString(O object) {
     }
 
     /** Generate router config for deployment as string. */
-    public byte[] generateRouterConfig() {
+    public byte[] generateRouterConfig() throws IOException {
         Project project = this.parentProject();
-        return customRouterConfig != null
-            ? customRouterConfig.getBytes(StandardCharsets.UTF_8)
-            : project.routerConfig != null
+
+        byte[] customRouterConfigString = customRouterConfig != null
+                ? customRouterConfig.getBytes(StandardCharsets.UTF_8) : null;
+
+        byte[] routerConfigString = project.routerConfig != null
                 ? writeToBytes(project.routerConfig)
                 : null;
+
+        // If both router configs are present, merge the JSON before returning
+        // Merger code from: https://stackoverflow.com/questions/35747813/how-to-merge-two-json-strings-into-one-in-java
+        if (customRouterConfigString != null && routerConfigString != null) {
+            ObjectMapper mapper = new ObjectMapper();
+            Map<String, Object> map1 = mapper.readValue(customRouterConfigString, Map.class);
+            Map<String, Object> map2 = mapper.readValue(routerConfigString, Map.class);
+            Map<String, Object> merged = new HashMap<String, Object>(map2);
+            merged.putAll(map1);
+            return mapper.writeValueAsString(merged).getBytes();
+        }
+
+        return customRouterConfigString != null
+            ? customRouterConfigString
+            : routerConfigString != null
+                ? routerConfigString
+                : null;
     }
 
     /** Generate router config for deployment as byte array (for writing to file output stream). */
     public String generateRouterConfigAsString() {
-        if (customRouterConfig != null) return customRouterConfig;
-        return writeToString(this.parentProject().routerConfig);
+            try {
+                return new String(this.generateRouterConfig(), "UTF-8");
+            } catch (IOException e) {
+                LOG.error("Failed to generate router config:");
```

Comment:
Combine these two lines to: `LOG.error("Failed to generate router config: ", e);`